### ISSUE TITLE
Optional Utterances with duplicated entities result

### ISSUE THREAD

**Issue by binsarjr** [axa-group/nlp.js#693]
When I process "2" the result I receive from optionalUtterances is "%number %% number%" is this normal or bug?

```{
  locale: 'id',
  utterance: '2',
  languageGuessed: false,
  localeIso2: 'id',
  language: 'Indonesian',
  nluAnswer: {...},
  classifications: [...],
  intent: 'None',
  score: 1,
  optionalUtterance: '%number%%number%',
  sourceEntities: [
    {
      start: 0,
      end: 0,
      resolution: [Object],
      text: '2',
      typeName: 'number'
    },
    {
      start: 2,
      end: 2,
      resolution: [Object],
      text: '2',
      typeName: 'number'
    },
  ],
  entities: [
    {
      start: 0,
      end: 0,
      len: 1,
      accuracy: 0.95,
      sourceText: '2',
      utteranceText: '2',
      entity: 'number',
      resolution: [Object],
      alias: 'number_0'
    },
    {
      start: 2,
      end: 2,
      len: 1,
      accuracy: 0.95,
      sourceText: '2',
      utteranceText: '2',
      entity: 'number',
      resolution: [Object],
      alias: 'number_1'
    },
  ],
  answers: [...],
  answer: ...,
  actions: [],
  sentiment: {...}
```

**Comment by jesus-seijas-sp** [comment]
Hello, is clearly a bug, confirmed, I am investigating, it seems a bug on the algorithm that reduces edges (basically, when there are several entities overlaped, there is an algorithm to remove some of them).

**Comment by jesus-seijas-sp** [comment]
Fixed by https://github.com/axa-group/nlp.js/pull/694
I will publish a version ASAP

**Comment by jesus-seijas-sp** [comment]
Published as 4.14.3
